### PR TITLE
Fixes #19630 - ignore hidden admins

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: test/models/notification_test.rb [hunk 1]
```diff
@@ -78,12 +78,15 @@ class NotificationTest < ActiveSupport::TestCase
...
     notification = FactoryGirl.build(:notification,
                                      :audience => Notification::AUDIENCE_ADMIN)
-    assert User.only_admin.count > 0
-    assert_equal User.only_admin.reorder('').pluck(:id).sort,
-      notification.subscriber_ids.sort
+    FactoryGirl.create(:user, :admin)
+
+    subscriber_ids = notification.subscriber_ids
+    assert_includes subscriber_ids, admin.id
+    assert_includes subscriber_ids, User.anonymous_admin.id
```
Comment (ohadlevy) [inline test/models/notification_test.rb]: I might be reading this wrong, but it sounds like the notification does notify the anonymous_admin(hidden user)? shouldnt it not include it?

File: test/models/notification_test.rb [hunk 2]
```diff
@@ -78,12 +78,15 @@ class NotificationTest < ActiveSupport::TestCase
...
+  test 'Admin notifications should subscribe to all admin users except hidden' do
     notification = FactoryGirl.build(:notification,
                                      :audience => Notification::AUDIENCE_ADMIN)
-    assert User.only_admin.count > 0
-    assert_equal User.only_admin.reorder('').pluck(:id).sort,
-      notification.subscriber_ids.sort
+    FactoryGirl.create(:user, :admin)
+
+    subscriber_ids = notification.subscriber_ids
+    assert_includes subscriber_ids, admin.id
```
Comment (ohadlevy) [inline test/models/notification_test.rb]: admin undefined?

**Comment by ares** [comment]
Thanks @ohadlevy, I must have pushed wrong version, I pushed the correct one, could please take a look?

**Comment by ohadlevy** [comment]
@ares ack pending tests...

**Comment by ohadlevy** [comment]
test failures are un related - thanks @ares

**Pull request merged**